content/flux/v0.x/data-types/dynamic.md

```diff
@@ -0,0 +1,622 @@
+menu:
+  flux_0_x:
+    name: Dynamic types
+    parent: Work with Flux types
+weight: 101
+flux/v0.x/tags: [dynamic types, data types]
+---
+
+A **dynamic** type is a wrapper for data whose type is not known until runtime.
+Dynamic types help when working with data from external sources (like JSON) 
+that support types that do not have an equivalent Flux type.
+
+- [Parse JSON into Flux types](#parse-json-into-flux-types)
+- [Dynamic type syntax](#dynamic-type-syntax)
+- [Reference values in a dynamic type](#reference-values-in-a-dynamic-type)
+- [Operate on dynamic types](#operate-on-dynamic-types)
+
+## Parse JSON into Flux types
+
+The primary (but not exclusive) use case for dynamic types is converting JSON
+data into native Flux types; specifically Flux [arrays](/flux/v0.x/data-types/composite/array/)
+and [records](/flux/v0.x/data-types/composite/record/).
+
+Because of strict typing rules in Flux, JSON data doesn't always gracefully parse
+into native Flux types.
+The most common reason this fails is because Flux arrays require all children
+to be the same type. JSON arrays allow elements to be different types.
+
+With Flux records (the corollary of a JSON object), the properties in the record
+are part of the record's type.
+If you have a JSON array of objects where the objects' schemas are different,
+the array cannot be parsed as a Flux array because the types of the records in
+the array do not match.
+
+{{< expand-wrapper >}}
+{{% expand "View examples of JSON arrays that cannot be directly parsed into Flux arrays" %}}
+
+#### Array of scalar values
+
+The following array is a valid JSON array, but an invalid Flux array because
+each element in the array is a different type.
+
+```json
+["three", 2, true]
+```
+
+#### Array of objects
+
+The following array is a valid JSON array, but an invalid Flux array.
+The schemas of the objects in the array do not match, so when converted to Flux
+records, their types do not match and the array is invalid.
+
+```json
+[
+    {"first-name": "John", "last-name": "Doe", "age": 42},
+    {"first-name": "Jane", "last-name": "Doe"},
+    {"first-name": "James", "last-name": "Doe", "age": 15}
+]
+```
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+The dynamic type provides a way to resolve the differences between JSON and Flux types.
+
+## Dynamic type syntax
+
+Flux does not provide a literal syntax for dynamic types.
+To cast a value to a dynamic type:
+
+1.  Import the [`experimental/dynamic` package](/flux/v0.x/stdlib/experimental/dynamic/).
+2.  Use [`dynamic.dynamic()`](/flux/v0.x/stdlib/experimental/dynamic/dynamic/) to
+    convert a value to a dynamic type.
+
+```js
+import "experimental/dynamic"
+
+dynamic.dynamic(v: "Example string")
+
+// Returns dynamic(Example string)
+```
+
+## Reference values in a dynamic type
+
+When working dynamic records and arrays, use bracket and dot notation to reference
+values in each.
+
+##### Reference values in a dynamic record
```

Comment:
With the change made (right at the end) to make attempted member access on a `dynamic(null)` an error, I might mention using `exists` with dynamic objects.

`exists` will "see through" the wrapper to check if there's a null inside or not, allowing us to guard against invalid access.